worker/peergrouper/worker.go

```diff
@@ -580,6 +597,8 @@ func (w *pgWorker) updateReplicaSet() (map[string]*replicaset.Member, error) {
 	if err != nil {
 		return nil, errors.Annotate(err, "creating peer group info")
 	}
+	// Update the metrics collector with the replicaset statuses.
+	w.metrics.update(info.statuses)
```

Comment:
Shouldn't this be a defer - we update the replicaset status below in the course of implementing this method and getting the status at the start of doing stuff won't really reflect the final status. Using a defer here means we'll still update the status even if this method exists in error.

Reply:
No, I don't think so. At this point in the code we have the information about the replicaset status. We just update our metric information as soon as we have it. We are recording the underlying replicaset status, not peergrouper work.